src/test/server/database/test_database.py

```diff
@@ -0,0 +1,159 @@
+        time.sleep(1)
+        self.job.status = JobStatus.PAUSED
+        self.mockDatabase.update_job(self.job)
+        time.sleep(1)
+        self.job.status = JobStatus.RUNNING
+        self.mockDatabase.update_job(self.job)
+        time.sleep(1)
+        self.job.status = JobStatus.DONE
+        self.mockDatabase.update_job(self.job)
+        j_entry = self.mockDatabase.find_job_entry(self.job.uid)
+        self.assertTrue(j_entry.stats.paused_time == 1)
+        self.assertTrue(j_entry.stats.running_time == 2)
+
+    def test_start_time(self) -> None:
+        self.job2.status = JobStatus.RUNNING
+        # add job to the database
+        self.mockDatabase.update_job(self.job)
+        # update with job2
+        self.mockDatabase.update_job(self.job2)
+        job_entry = self.mockDatabase.find_job_entry(self.job2.uid)
+        self.assertFalse(job_entry.stats.time_started is None)
+        self.assertFalse(job_entry.stats.time_added == job_entry.stats.time_started)
+
+    def test_from_dict_to_database(self) -> None:
+        object_dict = {
+            "uid": "bc",
+            "status": 0,
+            "owner_id": 1008,
+            "email": "user@website.com",
+            "scheduling_constraints": {"priority": 1, "is_preemptible": False, "special_resources": ["THING"]},
+            "docker_context": {
+                "dockerfile_source": "ssh localhost",
+                "mount_points": [{"source_path": "/home/user", "mount_path": "/home/user"}]
+            },
+            "docker_constraints": {"cpu_threads": 4, "memory": 4096},
+            "label": "thing"
+        }
+        job3 = Job.from_dict(object_dict)
+        self.mockDatabase.update_job(job3)
+        self.assertTrue(job3 == self.mockDatabase.find_job_by_id(job3.uid))
+
+    def test_update_work_machine(self) -> None:
+        self.mockDatabase.update_work_machine(self.work_machine)
+        self.assertTrue(self.mockDatabase.get_work_machines()[0] == self.work_machine)
+        work_machine2 = WorkMachine("machi1", WorkMachineState.OFFLINE,
+                                    WorkMachineResources(ResourceAllocation(12, 32, 12)))
+        self.mockDatabase.update_work_machine(work_machine2)
+        self.assertTrue(self.mockDatabase.get_work_machines()[0] == work_machine2)
+
+    def test_assign_job_machine(self) -> None:
+        self.mockDatabase.update_work_machine(self.work_machine)
+        self.mockDatabase.update_job(self.job)
+        self.mockDatabase.assign_job_machine(self.job, self.work_machine)
+        self.assertTrue(self.mockDatabase.get_jobs_on_machine(self.work_machine) == [self.job])
+
+    def test_schedule(self) -> None:
+        self.job2.status = JobStatus.KILLED
+        self.job2.uid = "job2"
+        self.job.status = JobStatus.RUNNING
+        self.mockDatabase.update_job(self.job)
+        self.mockDatabase.update_job(self.job2)
+        schedule = self.mockDatabase.get_current_schedule()
+        self.assertTrue(schedule, [self.job2])
```

Comment:
I think the logic for this check is broken.

Reply:
why

Reply:
Because calling self.assertTrue will evaluate schedule as a boolean and [self.job2] as an error message.

Reply:
Thank you I changed every assertTrue with Equal now, didn't now that it was evaluated as boolean